tests/integration/test_compute_flow.py

```diff
@@ -279,120 +350,139 @@ def test_compute_registered_algo(
     )
 
 
-@pytest.mark.skip(reason="TODO: reinstate integration tests")
 def test_compute_multi_inputs(
     publisher_wallet,
     publisher_ocean_instance,
     consumer_wallet,
     dataset_with_compute_service,
+    algorithm,
+    dataset_with_access_service,
 ):
     """Tests that a compute job with additional Inputs (multiple assets) starts properly."""
-    # Another dataset, this time with download service
-    another_dataset = get_registered_asset_with_access_service(
-        publisher_ocean_instance, publisher_wallet
+    run_compute_test(
+        ocean_instance=publisher_ocean_instance,
+        publisher_wallet=publisher_wallet,
+        consumer_wallet=consumer_wallet,
+        dataset_and_userdata=AssetAndUserdata(dataset_with_compute_service, None),
+        algorithm_and_userdata=AssetAndUserdata(algorithm, None),
+        additional_datasets_and_userdata=[
+            AssetAndUserdata(dataset_with_access_service, {"test_key": "test_value"})
+        ],
+        with_result=True,
     )
-    # verify the ddo is available in Aquarius
-    _ = publisher_ocean_instance.assets.resolve(another_dataset.did)
 
-    # Setup algorithm meta to run raw algorithm
-    algorithm = get_registered_algorithm_ddo_different_provider(
-        publisher_ocean_instance, publisher_wallet
-    )
-    _ = publisher_ocean_instance.assets.resolve(algorithm.did)
 
+def test_compute_trusted_algorithm(
+    publisher_wallet,
+    publisher_ocean_instance,
+    consumer_wallet,
+    dataset_with_compute_service_and_trusted_algorithm,
+    algorithm,
+    algorithm_with_different_publisher,
+):
+    # Expect to pass when trusted algorithm is used
     run_compute_test(
         ocean_instance=publisher_ocean_instance,
         publisher_wallet=publisher_wallet,
         consumer_wallet=consumer_wallet,
-        dataset_and_userdata=dataset_with_compute_service,
-        algorithm_and_userdata=algorithm,
-        userdata={"test_key": "test_value"},
+        dataset_and_userdata=AssetAndUserdata(
+            dataset_with_compute_service_and_trusted_algorithm, None
+        ),
+        algorithm_and_userdata=AssetAndUserdata(algorithm, None),
+        with_result=True,
     )
 
+    # Expect to fail when non-trusted algorithm is used
+    with pytest.raises(ValueError):
```

Comment:
please use the raw response and retrieve the error as well. Assert that the contents of the error is what you assume i.e. the algo is not trusted.

Reply:
Agreed.  I ran out of time to finish it tonight, but I think this is possible with the `match` argument of `pytest.raises`. I added TODOs.